packages/dx-react-grid/src/plugins/column-chooser.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Template, TemplatePlaceholder, PluginContainer, TemplateConnector,
+} from '@devexpress/dx-react-core';
+
+import { getMessagesFormatter, columnChooserItems } from '@devexpress/dx-grid-core';
+
+const pluginDependencies = [
+  { pluginName: 'TableColumnVisibility' },
+  { pluginName: 'Toolbar' },
+];
+export class ColumnChooser extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      visible: false,
+    };
+
+    this.handleToggle = this.handleToggle.bind(this);
+    this.handleHide = this.handleHide.bind(this);
+    this.getButtonRef = this.getButtonRef.bind(this);
+  }
+  getButtonRef(button) {
+    this.button = button;
+  }
+  handleToggle() {
```

Comment:
Maybe we must change this name: handleToggle -> handleOpen. Because it is not toggle function.
We will have toggleOpen and toggleHide